compiler/rustc_middle/src/ty/visit.rs

```diff
@@ -258,12 +260,8 @@ struct LateBoundRegionsCollector {
 }
 
 impl LateBoundRegionsCollector {
-    fn new(tcx: TyCtxt<'tcx>, just_constrained: bool) -> Self {
```

Comment:
huh why did this not cause a lint?

Reply:
Ah because it turns out my commits weren't as cleanly separated as thought ^^' where an earlier commit adds this unused binding and a later one removes it again. Let me fix that real quick.